bats/tests/helpers/paths.bash

```diff
@@ -1,17 +1,64 @@
+# PATH_BATS_ROOT and PATH_BATS_HELPERS are already set by load.bash
+
+PATH_REPO_ROOT=$(
+    # We are running with `set -o errexit`, so no need for explicit error checking
+    # shellcheck disable=SC2164
+    cd "$PATH_BATS_ROOT/.."
+    pwd
+)
+
+set_path_resources() {
+    local system=$1
+    local user=$2
+    local dist=$3
+    local subdir=$4
+
+    if [ -z "$RD_LOCATION" ]; then
+        if [ -d "$system" ]; then
+            RD_LOCATION=system
+        elif [ -d "$user" ]; then
+            RD_LOCATION=user
+        elif [ -d "$PATH_REPO_ROOT/pkg/rancher-desktop" ]; then
+            # dist and npm only work when inside a repo clone
+            if [ -d "$dist" ]; then
+                RD_LOCATION=dist
+            else
+                RD_LOCATION=npm
+            fi
+        else
+            fatal "Could not locate Rancher Desktop installation"
+        fi
+    fi
+    if using_npm_run_dev; then
+        if is_windows; then
+            fatal "npm operation not yet implemented for Windows"
+        fi
+        PATH_RESOURCES="$PATH_REPO_ROOT/resources"
+    else
+        PATH_RESOURCES="${!RD_LOCATION}/${subdir}"
+    fi
+    if [ ! -d "$PATH_RESOURCES" ]; then
+        fatal "App resource directory '$PATH_RESOURCES' does not exist"
+    fi
+}
+
 if is_macos; then
     PATH_APP_HOME="$HOME/Library/Application Support/rancher-desktop"
     PATH_CONFIG="$HOME/Library/Preferences/rancher-desktop"
     PATH_CACHE="$HOME/Library/Caches/rancher-desktop"
     PATH_LOGS="$HOME/Library/Logs/rancher-desktop"
     PATH_EXTENSIONS="$PATH_APP_HOME/extensions"
-    if test -d "/Applications/Rancher Desktop.app"; then
-        PATH_EXECUTABLE="/Applications/Rancher Desktop.app/Contents/MacOS/Rancher Desktop"
-        PATH_RESOURCES="/Applications/Rancher Desktop.app/Contents/Resources/resources"
-    else
-        PATH_EXECUTABLE="$(dirname "${BASH_SOURCE[0]}")/../../../dist/mac-$(uname -m)/Rancher Desktop.app/Contents/MacOS/Rancher Desktop"
-        PATH_RESOURCES="$(dirname "${BASH_SOURCE[0]}")/../../../dist/mac-$(uname -m)/Rancher Desktop.app/Contents/Resources/resources"
-    fi
     LIMA_HOME="$PATH_APP_HOME/lima"
+
+    DIST_ARCH="mac"
```

Comment:
is `DIST_ARCH` used anywhere else? (Can it be `local`?)

Reply:
It is not used anywhere else. It can't be `local` because we are not inside a function and `local` doesn't limit visibility to an `if` clause.